File: contest_F.cpp
//
// Created by defjia on 18-4-28.
//
#include <cstdlib>
#include <stdio.h>
#include <iostream>
using namespace std;

long long int test(long long int t){
    if(t == 1) return 1;
    else if(t == 2) return 3;
    else if(t % 2 == 1) return 2 * t -1;
    //else if(t % 2 == 0) return test(t/2) + test(t/2 + 1) + (t - 2)/2 - 1;
    else if(t % 2 == 0) return test(t/2 + 1) * 2 -2;
}
int main(){
    int a;
    long long int tmp;
    if(scanf("%lld", &a) != 1) exit(0);
    while (a != EOF && a != '\n'&& a != ' '){
        printf("%lld\n", test(a));
        cin >> a;
    }
    return 0;
}

File: Python/Unknown/contest_F.py
def test(q):
   if q == 1: return 1
   if q == 2: return 3
   elif q % 2 == 1: return 2 * q - 1
   elif q % 2 == 0: return test(q/2 + 1) * 2 -2

a = input()
while not str(a) == '':
    # if not a: break
    print(int(test(int(a))))
    a = input()
